mica/starcheck/process.py

```diff
@@ -113,73 +104,80 @@ def get_new_starcheck_files(rootdir, mtime=0):
+        for st in starchecks
+        if os.path.getmtime(st) >= mtime
+    ]
+    starchecks_with_times = sorted(starchecks_with_times, key=itemgetter("mtime"))
     return starchecks_with_times
 
 
 def update(config=None):
     if config is None:
         config = DEFAULT_CONFIG
-    if config['starcheck_db']['dbi'] != 'sqlite':
+    if config["starcheck_db"]["dbi"] != "sqlite":
         raise ValueError("Only sqlite DBI implemented")
     last_starcheck_mtime = 0
-    if (not os.path.exists(config['starcheck_db']['server'])
-            or os.stat(config['starcheck_db']['server']).st_size == 0):
-        if not os.path.exists(os.path.dirname(config['starcheck_db']['server'])):
-            os.makedirs(os.path.dirname(config['starcheck_db']['server']))
-        db_sql = Path(__file__).parent / FILES['sql_def']
+    if (
+        not os.path.exists(config["starcheck_db"]["server"])
+        or os.stat(config["starcheck_db"]["server"]).st_size == 0
+    ):
+        if not os.path.exists(os.path.dirname(config["starcheck_db"]["server"])):
+            os.makedirs(os.path.dirname(config["starcheck_db"]["server"]))
+        db_sql = Path(__file__).parent / FILES["sql_def"]
         db_init_cmds = open(db_sql).read()
-        db = Ska.DBI.DBI(dbi='sqlite', server=config['starcheck_db']['server'])
+        db = Ska.DBI.DBI(dbi="sqlite", server=config["starcheck_db"]["server"])
         db.execute(db_init_cmds)
     else:
-        db = Ska.DBI.DBI(dbi='sqlite', server=config['starcheck_db']['server'])
+        db = Ska.DBI.DBI(dbi="sqlite", server=config["starcheck_db"]["server"])
         max_mtime = db.fetchone("select max(mtime) as mtime from starcheck_id")
         if max_mtime is not None:
-            last_starcheck_mtime = max_mtime['mtime']
+            last_starcheck_mtime = max_mtime["mtime"]
     # If a start time is explicitly requested, override 0 or last database value
-    if 'start' in config and config['start'] is not None:
-        last_starcheck_mtime = DateTime(config['start']).unix
-    starchecks_with_times = get_new_starcheck_files(config['mp_top_level'],
-                                                    mtime=last_starcheck_mtime)
+    if "start" in config and config["start"] is not None:
+        last_starcheck_mtime = DateTime(config["start"]).unix
+    starchecks_with_times = get_new_starcheck_files(
+        config["mp_top_level"], mtime=last_starcheck_mtime
+    )
 
     for st_with_time in starchecks_with_times:
-        st = st_with_time['file']
+        st = st_with_time["file"]
         logger.info("Attempting ingest of %s" % st)
         # get an existing id or insert a new one
         mp_dir = re.search(
-            '\/data\/mpcrit1\/mplogs(\/\d{4}\/\w{7}/\w{5}\/)starcheck.txt',
-            st).group(1)
-        idcheck = db.fetchone(
-            "select id from starcheck_id where dir = '%s'" % mp_dir)
+            r"/data/mpcrit1/mplogs(/\d{4}/\w{7}/\w{5}/)starcheck.txt", st
```

Comment:
It looks like the previous defect in the regex was at least benign.  Agree that functional testing for the tiny update is sufficient.  This process/ingest module is not tested by the unit tests.